cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/local/LocalStorageClient.java

```diff
@@ -0,0 +1,34 @@
+package com.linkedin.openhouse.cluster.storage.local;
+
+import com.linkedin.openhouse.cluster.storage.StorageClient;
+import com.linkedin.openhouse.cluster.storage.StorageType;
+import javax.annotation.PostConstruct;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocalFileSystem;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+/**
+ * The LocalStorageClient class is an implementation of the StorageClient interface for local
+ * storage. It uses an Apache Hadoop FileSystem to interact with the local file system.
+ */
+@Slf4j
+@Lazy
+@Component
+public class LocalStorageClient implements StorageClient<FileSystem> {
+
+  private FileSystem fs;
+
+  /** Initialize the LocalStorageClient when the bean is accessed for the first time. */
+  @PostConstruct
+  private synchronized void init() {
+    log.info("Initialized storage client for type: " + StorageType.LOCAL.getValue());
+    this.fs = new LocalFileSystem();
```

Comment:
What about initializing this filesystem object, with 

https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/fs/LocalFileSystem.html#initialize-java.net.URI-org.apache.hadoop.conf.Configuration-

with endpoint `file:///` and rootPath '<path_to_root_dir>`?

And validate things like endpoint startsWith file:/// and path_to_root_dir exists?

Reply:
+1. This should be initialized with the hadoop configuration.

Reply:
Agree, added. PTAL.